src/schoology-extractor/edfi_schoology_extractor/extract_facade.py

```diff
@@ -191,20 +191,21 @@ def run(arguments: MainArguments) -> None:
         )
         sys.exit(1)
 
-    for section_id in result_bucket["sections"]["SourceSystemIdentifier"].values:
-        _get_section_associations(facade, arguments.output_directory, section_id)
+    if result_bucket["sections"].shape[0] > 0:
```

Comment:
Note above that we didn't validate that `result_bucket["sections"]` actually contains some data. The sections in Schoology were all set for last fall, therefore no data were returned and this threw a bug. I expanded the section grading period to cover this spring as well so that I could fully exercise the application.